### DISCUSSION TITLE
fromDataURL error: ERR_INVALID_URL

### DISCUSSION THREAD

**Discussion by jerlabCGI** [szimek/signature_pad#719] [Q&A]
I am using fromDataURL function passing the string that I received from the toDataURL function and I receive ERR_INVALID_URL. First why is it called fromDataURL when clearly in the exeample you are not passing an URL and second why am I doing wrong, here is my code:

var signaturePad = new SignaturePad(canvas, {
        backgroundColor: 'rgb(250,250,250)'
    });  

signaturePad.fromDataURL("data:image/png;base64,iVBORw0KGgoA...)

thanks!

**Comment by UziTech** [comment]
Can you provide the actual code so we can reproduce it. 

A data url is not the same as a regular url. `"data:image/png;base64,...` is the start of a data url

**Reply by jerlabCGI** [reply]
Hi Tony,

thanks for the quick reply and for the clarification for the dataURL. Here is my code:

Getting signature data url and put it in a hidden field:

` if (!signaturePad.isEmpty()) {
            let data = signaturePad.toDataURL();
            $('#DirectorSignature').val(data);
  }`

Then take the data url I got from previous code and set it in the signature pad using fromDataURL:

`signaturePad.fromDataURL("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA&#x2B;IAAADCCAYAAADNXnjgAAAAAXNSR0IArs4c6QAAGSJJREFUeF7t3TGIXMUfB/AnCB4iGIhwgSs8SSSFRYqAFkJy4F2bIoVdrGwVFMTGFHKxSGVhKwg5q3Q2gsbGiIKBFAoWkiYiZxRSWAQJGvz/&#x2B;S3M/efef&#x2B;9u727vt2/f&#x2B;zw4drPZfTPzmUm47868eY/9/fff/2kcBAgQIECAAAECBAgQIECAQIrAY4J4irNCCBAgQIAAAQIECBAgQIDASEAQNxAIECBAgAABAgQIECBAgECigCCeiK0oAgQIECBAgAABAgQIECAgiBsDBAgQIECAAAECBAgQIEAgUUAQT8RWFAECBAgQIECAAAECBAgQEMSNAQIECBAgQIAAAQIECBAgkCggiCdiK4oAAQIECBAgQIAAAQIECAjixgABAgQIECBAgAABAgQIEEgUEMQTsRVFgAABAgQIECBAgAABAgQEcWOAAAECBAgQIECAAAECBAgkCgjiidiKIkCAAAECBAgQIECAAAECgrgxQIAAAQIECBAgQIAAAQIEEgUE8URsRREgQIAAAQIECBAgQIAAAUHcGCBAgAABAgQIECBAgAABAokCgngitqIIECBAgAABAgQIECBAgIAgbgwQIECAAAECBAgQIECAAIFEAUE8EVtRBAgQIECAAAECBAgQIEBAEDcGCBAgQIAAAQIECBAgQIBAooAgnoitKAIECBAgQIAAAQIECBAgIIgbAwQIECBAgAABAgQIECBAIFFAEE/EVhQBAgQIECBAgAABAgQIEBDEjQECBAgQIECAAAECBAgQIJAoIIgnYiuKAAECBAgQIECAAAECBAgI4sYAAQIECBAgQIAAAQIECBBIFBDEE7EVRYAAAQIECBAgQIAAAQIEBHFjgAABAgQIECBAgAABAgQIJAoI4onYiiJAgAABAgQIECBAgAABAoK4MUCAAAECBAgQIECAAAECBBIFBPFEbEURIECAAAECBAgQIECAAAFB3BggQIAAAQIECBAgQIAAAQKJAoJ4IraiCBAgQIAAAQIECBAgQICAIG4MECBAgAABAgQIECBAgACBRAFBPBFbUQQIECBAgAABAgQIECBAQBA3BggQIECAAAECBAgQIECAQKKAIJ6IrSgCBAgQIECAAAECBAgQICCIGwMECBAgQIAAAQIECBAgQCBRQBBPxFYUAQIECBAgQIAAAQIECBAQxI0BAgQIECBAgAABAgQIECCQKCCIJ2IrigABAgQIECBAgAABAgQICOLGAAECBAgQIECAAAECBAgQSBQQxBOxFUWAAAECBAgQIECAAAECBARxY4AAAQIECBAgQIAAAQIECCQKCOKJ2IoiQIAAAQIECBAgQIAAAQKCuDFAgAABAgQIECBAgAABAgQSBQTxRGxFESBAgAABAgQIECBAgAABQdwYIECAAAECBAgQIECAAAECiQKCeCK2oggQIECAAAECBAgQIECAgCBuDBAgQIAAAQIECBAgQIAAgUQBQTwRW1EECBAgQIAAAQIECBAgQEAQNwYIECBAgAABAgQIECBAgECigCCeiK0oAgQIECBAgAABAgQIECAgiBsD&#x2B;xb4&#x2B;uuvmytXrjR3795tzp8/35w7d270&#x2B;Oyzz&#x2B;77XD5AgAABAgQIECBAgACBoQkI4kPr8UO2N0L42tra2LNEEL9w4cLo55dffhmF8/YhrB&#x2B;yA3ycAAECBAgQIECAAIG5FxDE574Lcxuwvr7exM9hjzqQl9n0eO3UqVPN0tLSaLZ9eXl59BhHPC9Hea39&#x2B;rg61e8t76/P2T5/KTfeWz8v597ri4T4AmKv9xzWbqfPR9ntOscXJ6U&#x2B;8fdx1HUsr8V7yvN4T/3nuj31Z8vrpW/GedV13ek8R&#x2B;XhvAQIECBAgAABAgS6KiCId7VnOlqv3WbEp1Hlp556qnnw4ME0TpV2jnbQLSG2/WVDVCj8zp4927zwwgvNtWvXtlYNxOvli4f4YiL&#x2B;XB/jwvNeXw7s9EXGuC8KxgXvvb5QqD8z7guI9mvx57iM4ebNm6Oqx/MS/kub4&#x2B;/i9fZRVlfsVae0TlcQAQIECBAgQIAAgUMICOKHwBvqR6c1K94OkhEcX3zxxWZhYWHbrPJOIXTSUNae6Y1yx80EjwvQO81wjyu7vHeSGeJ2eB83ltqz1PGeSds8b2Nzpz4uKxYioMd7yvvi8dKlSyOPcZdAzFv71ZcAAQIECBAgQGBYAoL4sPp7aq0tQTZmbktIevjw4ShE1zOdpcA6&#x2B;JYwGe&#x2B;L0NrXcDk1bCf6P4GyDH9jY2P0d3WQj4BuXBk0BAgQIECAAAECXRYQxLvcO&#x2B;pGgMBEAnUwr6&#x2B;Vjy956h&#x2B;z5xNxehMBAgQIECBAgMARCwjiRwzs9AQI5AuUFRhxHX79/OLFi83t27e3KlRmzn/77bdmdXW1efvtt/Mrq0QCBAgQIECAAIHBCQjig&#x2B;tyDSYwXIEyW14up4hLK&#x2B;o9BK5evdq89dZbwwXScgIECBAgQIAAgRQBQTyFWSEECHRV4Lnnnms2NzdH1Ytl7Hfu3OlqVdWLAAECBAgQIECgJwKCeE86UjMIEDiYQPsuADdu3LAT&#x2B;8EofYoAAQIECBAgQGBCAUF8QihvI0CgnwKxNP3555/fatzx48ebe/fu9bOxWkWAAAECBAgQINAJAUG8E92gEgQIzFJgbW2tievFy7G4uNj8&#x2B;uuvs6ySsgkQIECAAAECBHosIIj3uHM1jQCByQQihEcYrw9L1Cez8y4CBAgQIECAAIH9Cwji&#x2B;zfzCQIEeiYQy9NXVla2Nm2L5tm4rWedrDkECBAgQIAAgQ4JCOId6gxVIUBgdgJmxWdnr2QCBAgQIECAwNAEBPGh9bj2EiAwViBmxV9//fVt14qfP3&#x2B;&#x2B;iSXqDgIECBAgQIAAAQLTFBDEp6npXAQIzLWAWfG57j6VJ0CAAAECBAjMjYAgPjddpaIECBy1QMyKr66uNvFYjsuXLzfx4yBAgAABAgQIECAwLQFBfFqSzkOAQC8Erl27NlqiXg6btvWiWzWCAAECBAgQINApAUG8U92hMgQIzFpg3LXiZsVn3SvKJ0CAAAECBAj0S0AQ71d/ag0BAlMQaF8rHrPiH3/8cRObtzkIECBAgAABAgQIHFZAED&#x2B;soM8TINBLgfX19SZ&#x2B;ynHixInmm2&#x2B;&#x2B;Gd1f3EGAAAECBAgQIEDgMAKC&#x2B;GH0fJYAgV4LtK8XjxnxmBkXxnvd7RpHgAABAgQIEDhyAUH8yIkVQIDAPAtcv369effdd5vNzc1RMyKEf/XVV8L4PHequhMgQIAAAQIEZiwgiM&#x2B;4AxRPgED3BeKa8dhJvb6tWcyMv/baa92vvBoSIECAAAECBAh0TkAQ71yXqBABAl0UGBfGI4jHjuqWqnexx9SJAAECBAgQINBdAUG8u32jZgQIdEwgZsRXV1e3zYxHFcuO6gJ5xzpMdQgQIECAAAECHRUQxDvaMapFgEB3BdbW1pqYIa&#x2B;P2Mjtvffec4uz7nabmhEgQIAAAQIEOiMgiHemK1SEAIF5Eogd1eP2ZvV141F/y9XnqRfVlQABAgQIECAwGwFBfDbuSiVAoCcC4wJ5LFEvy9V70kzNIEBgDgQ&#x2B;/PDD5sGDB6MvBF0qMwcdpooECAxaQBAfdPdrPAEC0xIYF8jffPPN5o033vAL8bSQnYfAAATi/5I4YrVNWXFTr7x5&#x2B;PBhEz/Hjh1r7t69O3rvo0ePmnv37jX//vvvltA777zTfPDBBwMQ00QCBAjMp4AgPp/9ptYECHRQIH5ZjuXqX3zxRfPHH3&#x2B;MahjXjsfsuNmpDnaYKhHogEDcGvH3339vvvzyy6nXZmlpqXn//ffdanHqsk5IgACBwwsI4oc3dAYCBAhsExh3q7MbN27YyM04IUBgSyBmviOEH&#x2B;RYWFgYzYpPethMclIp7yNAgECegCCeZ60kAgQGJhC/ZJdlptH0uOe4azcHNgg0l0AlUJaYr6ysNJubm9tsYvb68ccfb5aXl5tz586NVtHET/lMPI&#x2B;/i6O9wia&#x2B;/Pvhhx&#x2B;ajY2N0eNOx&#x2B;LiYhMhvvw/FAHdah1DlAABArMREMRn465UAgQGItC&#x2B;dtxS9YF0vGYOTqAE5nJrw3KNdzyWa7nj&#x2B;ZNPPtn89ddf23w&#x2B;/fTT5tVXX52K2fXr15tbt241n3322f/d1WFcASWIx/9N8QVAHAL6VLrCSQgQILCrgCBugBAgQOCIBeKX79XV1W0zW3ZVP2J0pydwBALxb7nMUkfgjsAaj/F6hNh69rp9a8NSnW&#x2B;//bb5/PPPR/tIxMx0/F9wFEeUH3WLWfLy5cB&#x2B;yzl58mTz8ssvjz5WZujL83iMGfqDzKgXx/3Wx/sJECDQJwFBvE&#x2B;9qS0ECHRaYG1tbdsvxGbHO91dKjdwgTq8RnCMQBthu8wa77RMfBK2OPdBQ&#x2B;wk52&#x2B;/p56tv3nz5ugLg4OG8/a56yA&#x2B;ziTKiuvZf/rpp&#x2B;b48eNbX1b4/&#x2B;8gPekzBAj0SUAQ71NvagsBAp0XiOvGywxaqezFixebq1evHmhmqfMNVkECHRSoZ6vbtwsrM7/xGGExjoPM&#x2B;naw2duqVJbMx2OE8ziKSyylj&#x2B;fPPPNMc//&#x2B;/SNrSqwGiFUBDgIECAxRQBAfYq9rMwECMxWIX3BLIC8VKRsxxWPMuLlGc6ZdpPAeCZSZ3wibJXDGa&#x2B;1/c3Xw7lHzD9WUdjAvJxsX3EuQP3369GgGfKel&#x2B;XWF7ty508svOQ6F7sMECAxGQBAfTFdrKAECXROITZU&#x2B;&#x2B;uij5vvvvx9btdhBOXZSjt3WhYSu9Z76dEmgXnpdZnjLrG69GVn5oitzWXiXnLLrUofx6I9wb38Jkl0n5REgQKArAoJ4V3pCPQgQGKxALI2dZEOlEsYvXbrknuSDHS3DbngJdmU5eczM1tc6l38jZXm5Zc/DHi9aT4AAgS4LCOJd7h11I0BgUAL1BkoxU77b/YCF8kENjcE1dtwMdztwB0q5H7YVI4MbIhpMgACBuRcQxOe&#x2B;CzWAAIG&#x2B;CpTNlGK2vMwAjmtrhJAIJOXa8r56aFc/BdqhO8Z6fRuwaHXZM8EY7&#x2B;cY0CoCBAgMUUAQH2KvazMBAnMpEOEkQkodVNoNEcrnsmsHU&#x2B;ny5VLZOK1cxx0AZfO0CNtlHA8GRkMJECBAYHACgvjgulyDCRCYd4H6Otn19fUdm1OW69qFfd57fP7qX4/ReF5fdlE2T7OsfP76VY0JECBAYHoCgvj0LJ2JAAEC6QL18vU67IybKY/Xzpw501y4cGH0126Rlt5dvSuwvr3VTrPcZcO0eiO13kFoEAECBAgQ2KeAIL5PMG8nQIBAlwVKGI9QtFswL21o7zIdr9v4qss9PNu61XsVxGqM&#x2B;lruuDVVWVbuS57Z9pPSCRAgQKD7AoJ49/tIDQkQIHBggXrGMp7/&#x2B;OOPzZ9//rnr5m91SI9wVcJ5PEbQqv9c/r5&#x2B;7cCV9cFOCJS9COILmXJP7ieeeKL5&#x2B;eefR38uGwOWygrdneg2lSBAgACBORMQxOesw1SXAAEC0xKoZ8/jnOX2UCW877ecdiCL0H7q1Knmn3/&#x2B;aSKwl2uD93te75&#x2B;&#x2B;QPt&#x2B;3GUs1H1fb572yiuvNEtLS/pw&#x2B;l3hjAQIECAwUAFBfKAdr9kECBDYSaBcd16HsljqHkc7pNe7Xo8737Fjx0Yz8OUo4a6E8nEz7AL7dMdmvVN5e&#x2B;O0ul/ieXyZYnO/6fo7GwECBAgQGCcgiBsXBAgQIDAVgTqk1wG9LG8uhZSZ990KHRfY269NpdI9O0k9sx1fnuxmXXYtd2/ung0CzSFAgACBuRAQxOeim1SSAAEC/RGor1uPVtVBfdxs/LiW1xvKxfNYOh1L4GNGdyhHuZY72rtX6K435RO8hzJCtJMAAQIEuiwgiHe5d9SNAAECAxYooXxcWI/X6tneegn8yZMnm0ePHo02FTt9&#x2B;nTz0ksvzf21zdHWcnnAXqE7bCJ4lxlvm6kN&#x2B;B&#x2B;RphMgQIBAZwUE8c52jYoRIECAwG4C7dnzCKj17bXKZxcXF5uFhYXm8uXLo3Da5aN9Pfe49rTrX2&#x2B;qZra7y72rbgQIECBA4H8CgrjRQIAAAQK9Efjuu&#x2B;&#x2B;alZWVHdtTlmjPckOysjQ/ZrnrzdMm2a2&#x2B;bGRXZrvd8703Q1dDCBAgQGBgAoL4wDpccwkQINB3gbJh2ZUrV5rYNG63o&#x2B;wSHkvYb926NdrhvYTdeLx//35z9uzZrd3i22G5fu9OQbqE7ahHCd&#x2B;T9EE90x3Phe5J1LyHAAECBAjMh4AgPh/9pJYECBAgsE&#x2B;B&#x2B;j7pkyzx3ufpp/b2eiO1OKnl5VOjdSICBAgQINBZAUG8s12jYgQIECAwLYFZh/LYTO7pp59ulpeXR0G7Hb6n1U7nIUCAAAECBOZDQBCfj35SSwIECBCYokAsEd/Y2NjXUvFJi6&#x2B;v447PmOGeVM77CBAgQIDAcAQE8eH0tZYSIECAwBiBCOW3b99uPvnkk&#x2B;bEiRNb15XH89hx/cyZM3ve/qyE75jxLs9hEyBAgAABAgR2EhDEjQ0CBAgQIECAAAECBAgQIJAoIIgnYiuKAAECBAgQIECAAAECBAgI4sYAAQIECBAgQIAAAQIECBBIFBDEE7EVRYAAAQIECBAgQIAAAQIEBHFjgAABAgQIECBAgAABAgQIJAoI4onYiiJAgAABAgQIECBAgAABAoK4MUCAAAECBAgQIECAAAECBBIFBPFEbEURIECAAAECBAgQIECAAAFB3BggQIAAAQIECBAgQIAAAQKJAoJ4IraiCBAgQIAAAQIECBAgQICAIG4MECBAgAABAgQIECBAgACBRAFBPBFbUQQIECBAgAABAgQIECBAQBA3BggQIECAAAECBAgQIECAQKKAIJ6IrSgCBAgQIECAAAECBAgQICCIGwMECBAgQIAAAQIECBAgQCBRQBBPxFYUAQIECBAgQIAAAQIECBAQxI0BAgQIECBAgAABAgQIECCQKCCIJ2IrigABAgQIECBAgAABAgQICOLGAAECBAgQIECAAAECBAgQSBQQxBOxFUWAAAECBAgQIECAAAECBARxY4AAAQIECBAgQIAAAQIECCQKCOKJ2IoiQIAAAQIECBAgQIAAAQKCuDFAgAABAgQIECBAgAABAgQSBQTxRGxFESBAgAABAgQIECBAgAABQdwYIECAAAECBAgQIECAAAECiQKCeCK2oggQIECAAAECBAgQIECAgCBuDBAgQIAAAQIECBAgQIAAgUQBQTwRW1EECBAgQIAAAQIECBAgQEAQNwYIECBAgAABAgQIECBAgECigCCeiK0oAgQIECBAgAABAgQIECAgiBsDBAgQIECAAAECBAgQIEAgUUAQT8RWFAECBAgQIECAAAECBAgQEMSNAQIECBAgQIAAAQIECBAgkCggiCdiK4oAAQIECBAgQIAAAQIECAjixgABAgQIECBAgAABAgQIEEgUEMQTsRVFgAABAgQIECBAgAABAgQEcWOAAAECBAgQIECAAAECBAgkCgjiidiKIkCAAAECBAgQIECAAAECgrgxQIAAAQIECBAgQIAAAQIEEgUE8URsRREgQIAAAQIECBAgQIAAAUHcGCBAgAABAgQIECBAgAABAokCgngitqIIECBAgAABAgQIECBAgIAgbgwQIECAAAECBAgQIECAAIFEAUE8EVtRBAgQIECAAAECBAgQIEBAEDcGCBAgQIAAAQIECBAgQIBAooAgnoitKAIECBAgQIAAAQIECBAgIIgbAwQIECBAgAABAgQIECBAIFFAEE/EVhQBAgQIECBAgAABAgQIEBDEjQECBAgQIECAAAECBAgQIJAoIIgnYiuKAAECBAgQIECAAAECBAgI4sYAAQIECBAgQIAAAQIECBBIFBDEE7EVRYAAAQIECBAgQIAAAQIEBHFjgAABAgQIECBAgAABAgQIJAoI4onYiiJAgAABAgQIECBAgAABAoK4MUCAAAECBAgQIECAAAECBBIFBPFEbEURIECAAAECBAgQIECAAAFB3BggQIAAAQIECBAgQIAAAQKJAoJ4IraiCBAgQIAAAQIECBAgQICAIG4MECBAgAABAgQIECBAgACBRAFBPBFbUQQIECBAgAABAgQIECBAQBA3BggQIECAAAECBAgQIECAQKKAIJ6IrSgCBAgQIECAAAECBAgQICCIGwMECBAgQIAAAQIECBAgQCBRQBBPxFYUAQIECBAgQIAAAQIECBAQxI0BAgQIECBAgAABAgQIECCQKCCIJ2IrigABAgQIECBAgAABAgQICOLGAAECBAgQIECAAAECBAgQSBQQxBOxFUWAAAECBAgQIECAAAECBARxY4AAAQIECBAgQIAAAQIECCQKCOKJ2IoiQIAAAQIECBAgQIAAAQKCuDFAgAABAgQIECBAgAABAgQSBQTxRGxFESBAgAABAgQIECBAgAABQdwYIECAAAECBAgQIECAAAECiQKCeCK2oggQIECAAAECBAgQIECAgCBuDBAgQIAAAQIECBAgQIAAgUQBQTwRW1EECBAgQIAAAQIECBAgQEAQNwYIECBAgAABAgQIECBAgECigCCeiK0oAgQIECBAgAABAgQIECAgiBsDBAgQIECAAAECBAgQIEAgUUAQT8RWFAECBAgQIECAAAECBAgQEMSNAQIECBAgQIAAAQIECBAgkCjwXwQ1Z9X51geIAAAAAElFTkSuQmCC"); `

Thanks again!

**Reply by UziTech** [reply]
I see `&#x2B;` in your data url which should be a `+`. That makes it not a valid base64 string.
